src/backend/distributed/transaction/remote_transaction.c

```diff
@@ -1119,6 +1119,7 @@ CoordinatedRemoteTransactionsSavepointRollback(SubTransactionId subId)
 		}
 
 		FinishRemoteTransactionSavepointRollback(connection, subId);
+		UnclaimConnection(connection);
```

Comment:
`UnclaimConnection` means that we'd allow next commands to use this connection. So, I think it  might be useful to add a part of the PR description here as it might be confusing for users as this is an obvious point I guess:

>  We do not undo our hadDML and hadDDL
flags, when rolling back to a savepoint. This could result in some
queries not being allowed on Citus that would actually be fine to
execute. Changing this would require us to keep track for each
savepoint which placement connections had DDL/DML executed at that
point. 

Along with ` hadDML and hadDDL`, we probably should also mention that we do not `ResetShardPlacementAssociation` as well, which is also an interesting

Reply:
Done